core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -0,0 +1,270 @@
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.bookmarkUrlsForCurrentBook(zimFileReader).test()
+      .assertEmpty()
+  }
+
+  @Test
+  fun `bookmark urls for current workbook without zimFileReader`() {
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimName, "") } returns queryBuilder
+    every { queryBuilder.or() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.zimId, "") } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.bookmarkUrlsForCurrentBook(null).test()
+      .assertEmpty()
+  }
+
+  @Test
+  fun `saveBookmark saves the specific bookmark Item`() {
+    val bookmarkItem =
+      bookmarkItem(0L, "zimId", "zimName", "zimFilePath", "bookmarkUrl", "bookmarkTitle", "favicon")
+    newBookmarksDao.saveBookmark(bookmarkItem)
+    verify { box.put(BookmarkEntity(bookmarkItem)) }
+  }
+
+  @Test
+  fun `deleteBookmark removes query results for the url`() {
+    val bookmarkUrl = "bookmarkUrl"
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.equal(BookmarkEntity_.bookmarkUrl, bookmarkUrl) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    newBookmarksDao.deleteBookmark(bookmarkUrl)
+    verify { query.remove() }
+  }
+
+  @Test
+  fun `deleteBookmarks removes specified bookmark item list`() {
+    val id = 0L
+    val zimId = "zimId"
+    val zimName = "zimName"
+    val zimFilePath = "zimFilePath"
+    val bookmarkUrl = "bookmarkUrl"
+    val bookmarkTitle = "bookmarkTitle"
+    val favicon: String? = "favicon"
+    val bookmarkItem: BookmarkItem = mockk()
+    val bookmarksList: List<BookmarkItem> = listOf(bookmarkItem)
+    every { bookmarkItem.databaseId } returns id
+    every { bookmarkItem.zimId } returns zimId
+    every { bookmarkItem.zimName } returns zimName
+    every { bookmarkItem.zimFilePath } returns zimFilePath
+    every { bookmarkItem.bookmarkUrl } returns bookmarkUrl
+    every { bookmarkItem.bookmarkTitle } returns bookmarkTitle
+    every { bookmarkItem.favicon } returns favicon
+    newBookmarksDao.deleteBookmarks(bookmarksList)
+    verify {
+      box.remove(
+        listOf(
+          bookmarkEntity(
+            id = id,
+            zimId = zimId,
+            zimName = zimName,
+            zimFilePath = zimFilePath,
+            bookmarkUrl = bookmarkUrl,
+            bookmarkTitle = bookmarkTitle,
+            favicon = favicon
+          )
+        )
+      )
+    }
+  }
+
+  @Test
+  fun `migrate insert`() {
```

Comment:
how is any of the following code under test from this method?

```
  fun migrationInsert(
    bookmarks: MutableList<Bookmark>,
    bookDao: NewBookDao
  ) {
    box.put(bookmarks.zip(bookmarks.map(bookDao::getFavIconAndZimFile)).map(::BookmarkEntity))
  }
```

You need to be verifying `box.put(expectedEntities)`